invisible_cities/io/mcinfo_io.py

```diff
@@ -219,6 +219,10 @@ def read_mcinfo_evt (mctables: (tb.Table, tb.Table, tb.Table, tb.Table),
             while ipart <= ipart_end:
                 particle_rows.append(h5particles[ipart])
                 ipart += 1
+            
+            # It is possible for the 'generators' dataset to have a different length compared to the 'extents' dataset. In particular, it may occur that the 'generators' dataset is empty. In this case, do not add any rows to 'generators'.
+            if len(h5generators) == len(h5extents):
+                generator_rows.append(h5generators[last_row])
 
             # It is possible for the 'generators' dataset to have a different length compared to the 'extents' dataset. In particular, it may occur that the 'generators' dataset is empty. In this case, do not add any rows to 'generators'.
```

Comment:
This piece of code is duplicated. This is also the reason of the test failure.

Reply:
Ups you are right!